### PR TITLE
:memo: Getting started: OS X and DNX

### PR DESCRIPTION
**Pull request by natemcmaster**
#4 

Should be available here http://ef.readthedocs.org/en/staging/getting-started/osx.html

The sample likely have some build and execution errors until DNX renames are finished. 

Please help me check for clarity, grammar, etc.

### CODE REVIEW

File: docs/getting-started/osx.rst [hunk 1]
```diff
@@ -1,8 +1,187 @@
...
+-------------
+
+Minimum system requirements
+ - Mono 4.0.2
+ - OS X Mavericks
+
+.. note::
+    **Known Issues**
+
+     - Bugs in Mono 4.0.2 may cause Entity Framework to crash when using async methods. This is resolve with Mono >4.2.0, which has not yet been publicly released. `See this issue on GitHub <https://github.com/aspnet/EntityFramework/issues/2708>`_
```
Comment (ErikEJ) [inline docs/getting-started/osx.rst]: resolve => resolved

File: docs/getting-started/osx.rst [hunk 2]
```diff
@@ -1,8 +1,187 @@
...
+
+
+Begin using EF
+--------------
+With this new project, you are ready to begin using Entity Framework.
+We will create a simple console application that allows us to write a 
+blog post from the command line.
+
+ 1. Create a new class to represent the SQLite database. We will call this ``BloggingContext``. Note that to configure this for SQLite, we must call ``UseSqlite()`` with a connection string
+    pointing to the \*.db file.
```
Comment (ErikEJ) [inline docs/getting-started/osx.rst]: Forward slash or backslash?

Comment (natemcmaster) [inline docs/getting-started/osx.rst]: This is just escaping the * for rst files.

**Comment by natemcmaster** [comment]
Fixed some typos, updated formatting, and simplified the sample program.

Also, I made the installation commands more clear and added a final step to make it more clear how to execute the app.

File: docs/getting-started/osx.rst [hunk 3]
```diff
@@ -1,8 +1,230 @@
...
+    - `Install ASP.NET 5`_
+    - `Create a new project`_
+    - `Begin using EF`_
+    - `Create the database with migrations`_
+    - `Use the context to access data`_
+
+.. note:: `View this article's samples on GitHub <https://github.com/aspnet/EntityFramework.Docs/tree/master/docs/getting-started/x-plat/sample>`_.
+
+You can find nightly builds of the EF7 code base hosted on https://www.myget.org/F/aspnetvnext/api/v2/ but the code base is rapidly changing and we do not maintain up-to-date documentation for getting started.
+
```
Comment (rowanmiller) [inline docs/getting-started/osx.rst]: I think line 22 should be part of the note from line 9

File: docs/getting-started/osx.rst [hunk 4]
```diff
@@ -1,8 +1,230 @@
...
+Prerequisites
+-------------
+
+Minimum system requirements
+ - Mono 4.0.2
+ - OS X Mavericks
+
+.. note::
```
Comment (rowanmiller) [inline docs/getting-started/osx.rst]: use a `.. caution::`

File: docs/getting-started/osx.rst [hunk 5]
```diff
@@ -1,8 +1,230 @@
...
+          context    Commands to manage your DbContext
+          migration  Commands to manage your Code First Migrations
+          revEng     Command to reverse engineer code from a database
+          help       Show help information
+
+        Use "ef [command] --help" for more information about a command.
+
+
+Begin using EF
+--------------
```
Comment (rowanmiller) [inline docs/getting-started/osx.rst]: Rename title to "Create your model" (to align with other walkthroughs)

File: docs/getting-started/osx.rst [hunk 6]
```diff
@@ -1,8 +1,230 @@
...
+With this new project, you are ready to begin using Entity Framework.
+We will create a simple console application that allows us to write a 
+blog post from the command line.
+
+ - Create a new file called ``BloggingModel.cs``
```
Comment (rowanmiller) [inline docs/getting-started/osx.rst]: Call this Model.cs and just list the entire code in one block (to align with other walkthroughs).
http://ef.readthedocs.org/en/latest/getting-started/full-dotnet.html#create-your-model

File: docs/getting-started/osx.rst [hunk 7]
```diff
@@ -1,8 +1,230 @@
...
+
+        Build succeeded.
+            0 Warning(s)
+            0 Error(s)
+
+        Total build time elapsed: 00:00:01.9609581
+        Total projects built: 1
+
+Create the database with migrations
+-----------------------------------
```
Comment (rowanmiller) [inline docs/getting-started/osx.rst]: Rename title to "Create your database"

File: docs/getting-started/osx.rst [hunk 8]
```diff
@@ -1,8 +1,230 @@
...
+
+We can now use Entity Framework commands to create and manage the schema of our database.
+
+ - Create the first migration.
+    From the command line, execute ``dnx . ef commands migration add SampleInitial``.
+    This will find our context and models, and generate a migration for us in a folder named ``Migrations/``
+
+    .. code-block:: console
+
+        $ dnx . ef migration add SampleInitial
```
Comment (rowanmiller) [inline docs/getting-started/osx.rst]: Call it "MyFirstMigration" to align with other walkthroughs

File: docs/getting-started/osx.rst [hunk 9]
```diff
@@ -1,8 +1,230 @@
...
+    You can now using the existing migration to create the database file and creates the tables.
+
+    .. code-block:: console
+
+        $ dnx . ef migration apply
+
+    This should create a new file, ``blog.db`` which contains two empty tables.
+
+Use the context to access data
+------------------------------
```
Comment (rowanmiller) [inline docs/getting-started/osx.rst]: Rename to "Use your model"

File: docs/getting-started/x-plat/sample/src/ConsoleApp/Program.cs [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+                newPost.Content = sb.ToString();
+
+                context.Posts.Add(newPost);
+                context.SaveChanges();
+
+                Console.WriteLine("New post '${ newPost.Title }' saved!");
+            }
+        }
+    }
+}
```
Comment (rowanmiller) [inline docs/getting-started/x-plat/sample/src/ConsoleApp/Program.cs]: This code block should be consistent with the other walkthroughs (unless there is a need to change it for OSX).

```
            using (var db = new BloggingContext())
            {
                db.Blogs.Add(new Blog { Url = "http://blogs.msdn.com/adonet" });
                var count = db.SaveChanges();
                Console.WriteLine("{0} records saved to database", count);

                Console.WriteLine();
                Console.WriteLine("All blogs in database:");
                foreach (var blog in db.Blogs)
                {
                    Console.WriteLine(" - {0}", blog.Url);
                }
            }
```

**Comment by rowanmiller** [comment]
:shipit:

**Comment by rowanmiller** [comment]
BTW the commit emoji should be `:memo:` not :menu: :smile:

**Comment by natemcmaster** [comment]
Responded to comments and merged

**Pull request merged**